workspaces/adventure-pack/goodies/typescript/Map.groupBy/index.test.ts

```diff
@@ -0,0 +1,189 @@
+import { describe, expect, it } from "@jest/globals";
+delete (Map as any).groupBy;
```

Comment:
If we want to avoid `any`, we could tell TS to think of the `Map` object as a dictionary.

```ts
delete (Map as Record<string, unknown>).groupBy;
```

Sometimes TS is unwilling to accept a cast as plausible, if that happens, the trick is to go through `unknown`:

```ts
delete (Map as unknown as Record<string, unknown>).groupBy;
```

That's what we ended up doing for the iterator prototype stuff. In the end we are being a bit naughty, of course, but that's fine in a test.

Reply:
Oh I see. Yeah this is better, should have inferred what we were doing here from the iterator stuff because I definitely referenced it a lot.